var/spack/repos/builtin/packages/git/package.py

```diff
@@ -6,8 +6,10 @@
 import os
 import re
 
+from llnl.syscmd import is_system_path
+
+import spack.fetch_strategy
```

Comment:
The import system in `python` is flawed (personal opinion), and you might get away with not importing required packages, if they are transitively imported by something else. This is similar to transitive:
```
#include "xxxxx"
```
that you get from header files in C or C++. Since I polished the circular imports of a Python package in this PR, this revealed a few places where "xxxxx" was needed and not explicitly imported.

Reply:
I *personally* would vote for putting this sort of "fixes" into its own "cleanup / fixup / improve" PR. Something like "We learned, that more imports are really needed".

As I haven't done any spack tool development in a long time, please consider this only as something to possibly "ponder".